addons/sale_timesheet/models/project_update.py

```diff
@@ -26,37 +26,30 @@ def _get_services_values(self, project):
         if not project.allow_billable:
             return {}
         services = []
-        total_sold_per_uom = defaultdict(float)
-        total_effective_per_uom = defaultdict(float)
         total_sold, total_effective, total_remaining = 0, 0, 0
         sols = self._get_project_sols(project)
         for sol in sols:
             #We only want to consider hours and days for this calculation
             if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id:
+                name = sol.name_get()[0][1] if len(sols.order_id) > 1 else sol.name
+                product_uom_qty = round(sol.product_uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False), 1)
+                qty_delivered = round(sol.product_uom._compute_quantity(sol.qty_delivered, self.env.company.timesheet_encode_uom_id, raise_if_failure=False), 1)
                 services.append({
-                    'name': sol.name,
-                    'uom': sol.product_uom.name,
-                    'sold_value': sol.product_uom_qty,
-                    'effective_value': sol.qty_delivered,
-                    'remaining_value': sol.product_uom_qty - sol.qty_delivered,
+                    'name': name,
+                    'sold_value': product_uom_qty,
+                    'effective_value': qty_delivered,
+                    'remaining_value': product_uom_qty - qty_delivered,
                     'sol': sol,
                 })
-                total_sold_per_uom[sol.product_uom] += sol.product_uom_qty
-                total_effective_per_uom[sol.product_uom] += sol.qty_delivered
-        total_sold = sum(
-            uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
-            for uom, quantity in total_sold_per_uom.items()
-        )
-        total_effective = sum(
-            uom._compute_quantity(sol.product_uom_qty, self.env.company.timesheet_encode_uom_id, raise_if_failure=False)
-            for uom, quantity in total_effective_per_uom.items()
-        )
+        total_sold = sum(service['sold_value'] for service in services)
+        total_effective = sum(service['effective_value'] for service in services)
```

Comment:
You do 3 times the loop for the same elements, you could directly do it in the main for loop since we have all the data. 😇